test/test_Times.cc

```diff
@@ -0,0 +1,71 @@
+#include "gtest/gtest.h"
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include "../generic.h"
+#include "../SW_Sky.h"
+#include "../filefuncs.h"
+#include "../myMemory.h"
+#include "../SW_Defines.h"
+#include "../SW_Files.h"
+#include "../SW_Model.h"
+#include "../SW_Site.h"
+#include "../SW_SoilWater.h"
+#include "../SW_VegProd.h"
+#include "../SW_Site.h"
+#include "../SW_Flow_lib.h"
+#include "../Times.h"
+#include "sw_testhelpers.h"
+
+namespace{
+  // Test the 'Times.c' function 'interpolate_monthlyValues'
+  TEST(TimesTest, interpolateMonthlyValues){
+    // point to the structure that contains cloud coverage monthly values
+    SW_SKY SW_Sky;
+    SW_SKY *interpolate = &SW_Sky;
+    unsigned int i;
+    // set all monthlyValues all 10
+    for (i = 0; i < length(interpolate -> cloudcov); i++){
+      interpolate -> cloudcov[i] = 10;
+    }
+    interpolate -> cloudcov_daily[0] = 0;
+
+    // test various obtained values
+    interpolate_monthlyValues(interpolate -> cloudcov, interpolate -> cloudcov_daily);
+    // inperpolate_monthlyValues should not change index 0 because we used
+    // base1 indices
+    EXPECT_DOUBLE_EQ(interpolate -> cloudcov_daily[0], 0);
+    EXPECT_DOUBLE_EQ(interpolate -> cloudcov_daily[1], 10.0);
+    // test top conditional, day < 15
+    EXPECT_DOUBLE_EQ(interpolate -> cloudcov_daily[14], 10.0);
+    // test middle conditional, day >= 15
+    EXPECT_DOUBLE_EQ(interpolate -> cloudcov_daily[15], 10.0);
+    EXPECT_DOUBLE_EQ(interpolate -> cloudcov_daily[365], 10.0);
+    // Reset to previous global states
+    Reset_SOILWAT2_after_UnitTest();
+
+    // change first value to 20 and test the changes
+    interpolate -> cloudcov[0] = 20;
```

Comment:
Do you actually test these changes? Could you clarify your message to something like `change first value to a value to test if .....`

Reply:
Yes, changing cloudcov[0] translates to changing the cloud cover in January, which changes the interpolation of all the days. I test these changes underneath this initialization.